kms-provider-kroxylicious-inmemory/src/main/java/io/kroxylicious/kms/provider/kroxylicious/inmemory/InMemoryKms.java

```diff
@@ -0,0 +1,230 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.kms.provider.kroxylicious.inmemory;
+
+import java.security.GeneralSecurityException;
+import java.security.InvalidKeyException;
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.Supplier;
+
+import javax.crypto.Cipher;
+import javax.crypto.IllegalBlockSizeException;
+import javax.crypto.KeyGenerator;
+import javax.crypto.SecretKey;
+import javax.crypto.spec.GCMParameterSpec;
+
+import io.kroxylicious.kms.service.DekPair;
+import io.kroxylicious.kms.service.Kms;
+import io.kroxylicious.kms.service.KmsException;
+import io.kroxylicious.kms.service.Serde;
+import io.kroxylicious.kms.service.UnknownAliasException;
+import io.kroxylicious.kms.service.UnknownKeyException;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+
+/**
+ * An in-memory KMS to be used only for testing.
+ * Note that this exposes public methods that are not part of the {@link Kms} interface which are used for those
+ * KMS operations which are outside the scope of Kroxy itself (such as key provisioning).
```

Comment:
I want the `Kms` interface to be as minimal as possible for now. Adding operations to it means defining their semantics, which means coming to a good understanding of what is provided by the actual KMSes that we're abstracting over. Since the filter itself doesn't need key provisioning right now it's better to keep it out of the interface. We can abstract over key provisioning in tests using a `Supplier<K> provisionKey` if we're writing tests as that level of abstraction.